### DISCUSSION TITLE
Country "unknown"

### DISCUSSION THREAD

**Discussion by bernd-wechner** [umami-software/umami#1135] [General]
Really I just installed this and testing it on a number of sites I host. When I surf there I check to see it's registered, but oddly I am always detected as Country "unknown". I wonder how country is detected, and why it unreliably. Experience (without delving into the codebase) suggests the IP address reveals the country. 

https://www.whatismyip.com/

sees my country fine (though the city is a bit way out).

**Comment by mikecao** [comment]
See https://github.com/mikecao/umami/issues/814. It's possible you are being passed non-standard headers.

**Comment by bernd-wechner** [comment]
Thanks, I'll read through that and see what I can find. A question does remain: should Umami perhaps be robust against ordinary scenarios like this (premised on the notion that my experience is not exceptional). Or does the premise fail, am I in a very very rare exception not relevant scenario of sorts (as I said, I'll drill down later into https://github.com/mikecao/umami/issues/814 and see what I can find)

**Comment by mikecao** [comment]
Umami uses a package that tries to determine the IP address from lots of different headers. However, there are times when you are behind a proxy, have a misconfigured web server, etc. The solution is to figure out what header represents the client IP and then you can tell umami to always use that.
